### ISSUE TITLE
Kestral is very slow when posting data

### ISSUE THREAD

**Issue by cibinmathewjose** [aspnet/KestrelHttpServer#2979]
Please get the sample source from  https://github.com/cibinmathewjose/KestrelPostIssue.git 
run the code and post {"Name":"test","Price":1} to http://{machine}:5000/api/products, it will take ~6 seconds.

if you post {"Price":1} to http://{machine}:5000/api/products it will work normally.

Also if you use localhost instead of machine name in url, it will work normal. 
this is the same issue as #2229

**Comment by cibinmathewjose** [comment]
The put request endpoint is always slow with machine name url, regardless of the data posted with localhost url it is normal

**Comment by cibinmathewjose** [comment]
I don’t think so, My firewall is turned off. Also it doesn’t happen when there is no “Name” property in the JSON  payload for post method

**Comment by Tratcher** [comment]
When in doubt, simplify. Remove MVC from your sample and read the request body stream as text.

app.Run(async context =>
{
  var bytes = new byte[100];
  var read = await context.Request.Body.ReadAsync(bytes, 0, bytes.Length);
  Encoding.UTF8.GetString(bytes, 0, read);
  // etc..
});
